src/uu/ls/src/ls.rs

```diff
@@ -537,19 +539,45 @@ fn extract_time(options: &clap::ArgMatches) -> Time {
     }
 }
 
+/// Parse the dircolors terminal list into a RegexSet and check against
+/// current terminal if applicable
+///
+/// # Returns
+///
+/// Boolean
+fn check_env_variables_against_dircolors(dircolors: &str) -> bool {
+    let ls_colors = env::var("LS_COLORS").unwrap_or(String::new()).is_empty();
+    let colorterm = env::var("COLORTERM").unwrap_or(String::new()).is_empty();
+    if ls_colors && colorterm {
+        let term = env::var("TERM").unwrap_or(String::new());
+        let terminals = dircolors
+            .lines()
+            .filter(|x| x.starts_with("TERM "))
+            .map(|x| x.replace("TERM ", ""))
+            .map(|x| x.replace('*', ".*"))
+            .collect::<Vec<String>>();
+        let dircolors_set = RegexSet::new(terminals).unwrap();
+        dircolors_set.matches(term.as_str()).matched_any()
```

Comment:
Can't we use a glob parser directly, without having to convert to regex?

Reply:
@tertsdiepraam I will have a look and circle back

Reply:
I've moved the check to glob, it should be more consistent now